src/backend/utils/sort/tuplesort.c

```diff
@@ -3232,6 +3232,14 @@ dumptuples(Tuplesortstate *state, bool alltuples)
 		!alltuples)
 		return;
 
+	/*
+	 * do nothing if we are told to finish execution.
+	 */
+	if (QueryFinishPending)
```

Comment:
tuplecontext isReset is true ?

Reply:
tuplecontext is reset in above case.

Reply:
> which caller reset tuplecontext ?
In dumptuples, the tuplecontext is reset when all memtuples are consumed or QueryFinishPending is set to true.

Reply:
it means enter dumptuples twice ?